src/arrayMethodReduce.js

```diff
@@ -5,7 +5,15 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const initialValueTest = !(arguments.length < 2);
+    const startIndex = initialValueTest ? 0 : 1;
+    let accumulator = initialValueTest ? initialValue : this[0];
+
+    for (let i = startIndex; i < this.length; i++) {
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )